text/0000-lifecycle-all.md

```diff
@@ -0,0 +1,110 @@
+# Meta
+[meta]: #meta
+- Name: Lifecycle All Binary
+- Start Date: 2020-01-21
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+### `/cnb/lifecycle/all`
+A new lifecycle binary with the name `lifecycle/all` will be included in each released lifecycle archive. Within a builder image, it will be found at the path `/cnb/lifecycle/all`. When invoked, it will run the following phases sequentially:
+
+* detector
```

Comment:
Is it possible that a poorly written buildpack could mutate the workingDir during `detect` and those mutations would stick around through the other phases? Then when the same buildpack is run in Tekton, for example, with each phase in it's own container the output/results of buildpack execution against the same code are different?

Maybe there are other scenarios that don't involve a poorly written buildpack.

If so, is there any way to prevent that, such as running in phase in a sandbox (like a temp dir that we copy into during each phase)?

Reply:
Given that the working dir is `/layers` and both `/layers` and `/workspace` are volumes, changes made by buildpacks to these dirs are already persisted between phases. Changes made within `/buildpacks` wouldn't be persisted by the permissions should already be preventing buildpacks from making any such changes.